src/hooks/switch/cowprotocol.rates.ts

```diff
@@ -0,0 +1,168 @@
+import { OrderKind, QuoteAndPost, TradingSdk } from '@cowprotocol/cow-sdk';
+import { BigNumber } from 'bignumber.js';
+import {
+  COW_APP_DATA,
+  COW_PARTNER_FEE,
+  HEADER_WIDGET_APP_CODE,
+  isNativeToken,
+} from 'src/components/transactions/Switch/cowprotocol.helpers';
+import {
+  isChainIdSupportedByCoWProtocol,
+  WrappedNativeTokens,
+} from 'src/components/transactions/Switch/switch.constants';
+import { SwitchParams, SwitchRatesType } from 'src/components/transactions/Switch/switch.types';
+import { getEthersProvider } from 'src/libs/web3-data-provider/adapters/EthersAdapter';
+import { CoWProtocolPricesService } from 'src/services/CoWProtocolPricesService';
+import { getErrorTextFromError, TxAction } from 'src/ui-config/errorMapping';
+import { wagmiConfig } from 'src/ui-config/wagmiConfig';
+
+export async function getCowProtocolSellRates({
+  chainId,
+  amount,
+  srcToken,
+  srcDecimals,
+  destToken,
+  destDecimals,
+  user,
+  inputSymbol,
+  outputSymbol,
+  setError,
+}: SwitchParams): Promise<SwitchRatesType> {
+  const cowProtocolPricesService = new CoWProtocolPricesService();
+  const tradingSdk = new TradingSdk({ chainId });
+
+  let orderBookQuote: QuoteAndPost | undefined;
+  let srcTokenPriceUsd: string | undefined;
+  let destTokenPriceUsd: string | undefined;
+  try {
+    if (!isChainIdSupportedByCoWProtocol(chainId)) {
+      throw new Error('Chain not supported by CowProtocol');
+    }
+
+    // If srcToken is native, we need to use the wrapped token for the quote
+    let srcTokenWrapped = srcToken;
+    if (isNativeToken(srcToken)) {
+      srcTokenWrapped = WrappedNativeTokens[chainId];
+    }
+
+    let destTokenWrapped = destToken;
+    if (isNativeToken(destToken)) {
+      destTokenWrapped = WrappedNativeTokens[chainId];
+    }
+
+    const provider = await getEthersProvider(wagmiConfig, { chainId });
+    const signer = provider?.getSigner();
+
+    if (!inputSymbol || !outputSymbol) {
+      throw new Error('No input or output symbol provided');
+    }
+
+    [orderBookQuote, srcTokenPriceUsd, destTokenPriceUsd] = await Promise.all([
+      tradingSdk
+        .getQuote(
+          {
+            owner: user as `0x${string}`,
+            kind: OrderKind.SELL,
+            amount,
+            sellToken: srcTokenWrapped,
+            sellTokenDecimals: srcDecimals,
+            buyToken: destTokenWrapped,
+            buyTokenDecimals: destDecimals,
+            signer,
+            appCode: HEADER_WIDGET_APP_CODE, // todo: use ADAPTER_APP_CODE for contract adapters
+            partnerFee: COW_PARTNER_FEE(inputSymbol, outputSymbol),
+          },
+          {
+            appData: COW_APP_DATA(inputSymbol, outputSymbol),
+          }
```

Comment:
You don't need both, the info you want to include is already in the `partnerFee`

```suggestion
            partnerFee: COW_PARTNER_FEE(inputSymbol, outputSymbol),
          }
```